### PR TITLE
LlamaDb wrapper

### PR DESCRIPTION
**Pull request by RicoGit**
### Example of work with llamadb.

Contains a wrapper in Rust that allows run Llamadb in the Fluence WasmVm.
It provides 'allocate' and 'deallocate' methods for memory management (needed for passing string parameters to Wasm code).
It also contains helpers method for converting llamadb results into a string, writing a string to memory as raw bytes and read raw bytes as Rust string from the memory.

### CODE REVIEW

File: vm/examples/llamadb/src/main/rust/lib.rs [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+            col_names + &rows_as_str
+        }
+        ExecuteStatementResponse::Explain(result) => {
+            result.clone()
+        }
+    }
+}
+
+/// Writes Rust string into the memory directly as string length and byte array
+/// (big-endian order). Written memory structure is:
```
Comment (michailvoronov) [inline vm/examples/llamadb/src/main/rust/lib.rs]: little-endian?

File: vm/examples/llamadb/src/main/rust/lib.rs [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+///     | str_length: 4 BYTES | string_payload: str_length BYTES|
+unsafe fn put_to_mem(str: String) -> *mut u8 {
+    // converting string size to bytes in big-endian order
```
Comment (michailvoronov) [inline vm/examples/llamadb/src/main/rust/lib.rs]: little-endian?

File: vm/examples/llamadb/src/main/rust/lib.rs [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    let len_as_bytes: [u8; STR_LEN_BYTES] = mem::transmute((str.len() as u32).to_le());
+    let mut result: Vec<u8> = Vec::with_capacity(STR_LEN_BYTES + str.len());
+    result.write_all(&len_as_bytes).unwrap();
+    result.write_all(str.as_bytes()).unwrap();
+
+    let result_ptr = allocate(result.len()).as_ptr();
+
+    // writes bytes into memory byte-by-byte. Address of first byte will be == `ptr`
```
Comment (michailvoronov) [inline vm/examples/llamadb/src/main/rust/lib.rs]: Are there any C memcpy analog in Rust?

File: vm/examples/llamadb/src/main/rust/lib.rs [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    for (idx, byte) in result.iter().enumerate() {
+        std::ptr::write(result_ptr.offset(idx as isize), *byte);
+    }
+
+    result_ptr
+}
+
+/// Creates a public static reference to initialized LlamsDb instance.
```
Comment (michailvoronov) [inline vm/examples/llamadb/src/main/rust/lib.rs]: Llam**a**Db

File: vm/examples/llamadb/src/main/rust/tests.rs [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+    // delete and update is not supported by llamadb
+
+}
+
+//
+// Private helper functions.
+//
+
+/// Execute sql and returns result as String.
```
Comment (michailvoronov) [inline vm/examples/llamadb/src/main/rust/tests.rs]: Executes

File: vm/examples/llamadb/src/main/rust/lib.rs (hunk 3 above)
Comment (RicoGit) [reply vm/examples/llamadb/src/main/rust/lib.rs]: libc

File: vm/examples/llamadb/src/main/rust/lib.rs [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+/// Result for all possible Error types.
+type GenResult<T> = Result<T, Box<Error>>;
+
+pub const STR_LEN_BYTES: usize = 4;
+
+//
+// Public functions for work with Llamadb.
+//
+
+/// Execute sql and returns result as string in the memory.
```
[the remark this answers is not in the document]
Comment (michailvoronov) [reply vm/examples/llamadb/src/main/rust/lib.rs]: Anyway, there are a lot of using "memory" and "the memory" in these PR, let come to any single variant.

**Pull request merged**